packages/@lwc/template-compiler/src/parser/html.ts

```diff
@@ -86,6 +86,58 @@ export function cleanTextNode(value: string): string {
 
     return str;
 }
+/**
+ * In order to support property interpolation within text nodes, we ignore
+ * how parse5 parses text nodes and instead grab the raw source string by
+ * looking at the start and end offsets.  However, if a non-comforming HTML
+ * tag is included in the template, that tag and its children will be
+ * contained within the slice of the raw source string.
+ *
+ * For example, given the following template:
+ *
+ *  <template>
+ *    <tr>
+ *      <unknown></unknown>
+ *    </tr>
+ *  </template>
+ *
+ * There will be a single child text node of the <tr> tag.  It's value will
+ * be a handful of whitespace characters.  The slice of the original source
+ * using start and end offsets will include whitespace plus
+ * "<unknown></unknown>" as a raw string.
+ *
+ * This function strips out the unparsed non-conforming HTML.
+ */
+export function omitNonconformingTags(node: parse5.AST.Default.TextNode, sourceSubString: string) {
+    const { startOffset, endOffset } = node.__location!;
+
+    const parent = node.parentNode as parse5.AST.Default.Element;
+    const grandparent = parent?.parentNode as parse5.AST.Default.Element;
+    let parentSibs = (grandparent?.childNodes as Array<parse5.AST.Default.Element>) || [];
+    parentSibs = parentSibs.slice().reverse();
```

Comment:
I am not sure if checking the parent sibling is good enough. It appears to work in this case where an invalid tag is nested inside a `<tr>` element. However it doesn't work is a more complex case like:

```html
<template>
  <table>
    <tbody>
      <tr>
        <unknown></unknown>
      </tr>
    </tbody>
  </table>
</template>
```

I have been thinking about this specific issue in #2433. I think the safest thing we can do is to detect whenever HTML content is invalid and report a compilation warning. We can also report a compilation error if it is guarded behind a flag to avoid introducing a breaking change.

I see 2 cases where parse5 autocorrects invalid HTML:
- adding a missing element: `<table><tr></tr></table>` --> `<table><tbody><tr></tr></tbody></table>`. We can detect this case as no source location is attached to missing element.
- reshuffling elements: `<table><foo></foo></table>` --> `<foo></foo><table></table>`. We can also detect this by comparing each element offset with its sibling. If the offset are overlaping we can safely assume that the parser reshuffled the element.

Reply:
Thanks for pointing out the uncovered case.  It does seem pretty reasonable to address this by emitting compiler warnings, so I'll close this PR in favor of #2433.